backend/courses/models.py

```diff
@@ -433,6 +433,165 @@ def __str__(self):
         return f"Topic {self.id} ({self.most_recent.full_code} most recently)"
 
 
+class Attribute(models.Model):
+    """
+    A post-NGSS registration attribute, which is used to
+    mark courses which students in a program/major should take.
+    e.g. WUOM for the "Wharton OIDD Operation" track
+
+    Note that Attributes (like Restrictions) do not have an associated
+    semester. This is because only the Attribute/NGSSRestriction set used in the
+    current semester is maintained. Specifically, Attribute and NGSSRestriction
+    objects are regenerated at every upsert from OpenData such that every
```

Comment:
Might want to change this wording if we remove the delete from the `upsert_course_from_opendata` function.